Artifacts/windows-7zip/ChocolateyPackageInstaller.ps1

```diff
@@ -0,0 +1,262 @@
+﻿<##################################################################################################
+
+    Description
+    ===========
+
+	- This script does the following - 
+		- installs chocolatey
+		- installs specified chocolatey packages
+
+	- This script generates logs in the following folder - 
+		- %ALLUSERSPROFILE%\ChocolateyPackageInstaller-{TimeStamp}\Logs folder.
+
+
+    Usage examples
+    ==============
+    
+    Powershell -executionpolicy bypass -file ChocolateyPackageInstaller.ps1
+
+
+    Pre-Requisites
+    ==============
+
+    - Ensure that the powershell execution policy is set to unrestricted (@TODO).
+
+
+    Known issues / Caveats
+    ======================
+    
+    - No known issues.
+
+
+    Coming soon / planned work
+    ==========================
+
+    - N/A.    
+
+##################################################################################################>
+
+#
+# Optional arguments to this script file.
+#
+
+Param(
+    # comma or semicolon separated list of chocolatey packages.
+    [ValidateNotNullOrEmpty()]
+    [string]
+    $RawPackagesList
+)
+
+##################################################################################################
+
+#
+# Powershell Configurations
+#
+
+# Note: Because the $ErrorActionPreference is "Stop", this script will stop on first failure.  
+$ErrorActionPreference = "stop"
+
+###################################################################################################
+
+#
+# Custom Configurations
+#
+
+$ChocolateyPackageInstallerFolder = Join-Path $env:ALLUSERSPROFILE -ChildPath $("ChocolateyPackageInstaller-" + [System.DateTime]::Now.ToString("yyyy-MM-dd-HH-mm-ss"))
```

Comment:
It's probably best to replace $env:ALLUSERSPROFILE with $env:USERPROFILE since the latter location can be written to even by a non-admin.